packages/panzer/adapters-ioss/src/Panzer_IOSSConnManager.cpp

```diff
@@ -40,21 +40,970 @@
+#include "Panzer_FieldPattern.hpp"
+#include "Panzer_IOSSDatabaseTypeManager.hpp"
+
+// Zoltan includes
+#include "Zoltan2_findUniqueGids.hpp"
+
+// Ioss includes
+#include "Ioss_CodeTypes.h"
+#include "Ioss_ParallelUtils.h"
+#include "Ioss_DatabaseIO.h"
+#include "Ioss_GroupingEntity.h"
+#include "Ioss_Region.h"
+#include "Ioss_EntityBlock.h"
+#include "Ioss_ElementTopology.h"
+#include "Ioss_NodeBlock.h"
+#include "Ioss_ElementBlock.h"
 
 namespace panzer_ioss {
 
-template class IOSSConnManager<int>;
+using Teuchos::RCP;
+using Teuchos::rcp;
+
+IOSSConnManager::IOSSConnManager(Ioss::DatabaseIO * iossMeshDB)
+   : iossMeshDB_(iossMeshDB), iossRegion_(), iossNodeBlocks_(), iossElementBlocks_(),
+	 iossConnectivity_(), iossToShardsTopology_(), iossElementBlockTopologies_(),
+	 elementBlocks_(), neighborElementBlocks_(),
+	 numUniqueEdges_(0), numUniqueFaces_(0), edgeNodeToEdgeMap_(), faceNodeToFaceMap_(),
+	 elementEdgeNodes_(), elementFaceNodes_(),
+	 elmtLidToGid_(), elmtLidToConn_(), connSize_(),
+	 connectivity_(), ownedElementCount_(0), sidesetsToAssociate_(),
+	 sidesetYieldedAssociations_(), elmtToAssociatedElmts_(), placeholder_()
+{
+
+  std::string error_message;
+  int bad_count;
+
+  // Check for a non-null database
+  TEUCHOS_TEST_FOR_EXCEPTION(iossMeshDB_ == nullptr, std::logic_error, "Error, iossMeshDB is nullptr.");
+  TEUCHOS_TEST_FOR_EXCEPTION(iossMeshDB_ == NULL, std::logic_error, "Error, iossMeshDB is NULL.");
+
+  // Check that the database is OK and that the file can be opened.
+  bool status_ok = iossMeshDB_->ok(true, &error_message, &bad_count);
+  TEUCHOS_TEST_FOR_EXCEPTION(!status_ok, std::logic_error,
+		  "Error, Ioss::DatabaseIO::ok() failed.\n error_message = "
+		  << error_message << "\nbad_count = " << bad_count << "\n");
+
+  // Get the metadata
+  iossRegion_ = rcp(new Ioss::Region(iossMeshDB_, "iossRegion_"));
+  iossNodeBlocks_ = iossRegion_->get_node_blocks();
+  iossElementBlocks_ = iossRegion_->get_element_blocks();
+
+  // Get element block topologies
+  for (Ioss::ElementBlock * iossElementBlock : iossElementBlocks_) {
+    iossElementBlockTopologies_[iossElementBlock->name()] = iossElementBlock->topology();
+  }
+
+  // Create iossToShardsTopology_ map;
+  createTopologyMapping();
+
+}
+
+Teuchos::RCP<panzer::ConnManager> IOSSConnManager::noConnectivityClone(
+  std::string & type, Ioss::PropertyManager & properties) const {
+	   std::string filename = iossMeshDB_->get_filename();
+
+	   // Verify that multiple open databases are supported
+	   TEUCHOS_TEST_FOR_EXCEPTION(IossDatabaseTypeManager::supportsMultipleOpenDatabases(type), std::logic_error, "Error, " << type << "does not support multiple open databases.");
+
+	   // temporarily try this with a _copy file
+	   size_t dotPosition = filename.find(".");
+	   std::string filenameBase(filename, 0, dotPosition);
+	   std::string filenameExtension(filename, dotPosition+1);
+	   std::string filenameCopy = filenameBase + "_copy." + filenameExtension;
```

Comment:
It looks like you might want to use POSIX's temp file creation facility here.  There's a way to put temp files in any directory you want and to give them any prefix name you want.